lib/Sema/TypeCheckAttr.cpp

```diff
@@ -863,7 +863,28 @@ void AttributeChecker::visitSetterAccessAttr(
   }
 }
 
-void AttributeChecker::visitSPIAccessControlAttr(SPIAccessControlAttr *attr) {}
+void AttributeChecker::visitSPIAccessControlAttr(SPIAccessControlAttr *attr) {
+  if (isa<ImportDecl>(D) || isa<ExtensionDecl>(D) || isa<TypeAliasDecl>(D))
+    return;
+
+  auto VD = dyn_cast<ValueDecl>(D);
+  if (!VD) {
+    diagnose(attr->getLocation(),
```

Comment:
Isn't this already handled as part of the definition in Attr.def?

Reply:
Right, we only need the check for the access level of the decl.